src/view/document.js

```diff
@@ -108,15 +108,49 @@ export default class Document {
 	 * a change, it should return `true`. When this happens, all post-fixers are fired again to check if something else should
 	 * not be fixed in the new document tree state.
 	 *
-	 * View post-fixers are useful when you wants to update view structure whenever it changes, for instance add some classes
-	 * to elements based on the view structure or selection. However, is you need DOM elements to be already updated, use
-	 * {@link module:engine/view/view~View#event:render render event}.
+	 * View post-fixers are useful when you wants to apply some fixes whenever view structure changes. Keep in mind that
+	 * changes executed in the view post-fixer may break model-view mapping, which may cause bugs. To
+	 * avoid them make sure that your post-fixer do only save changes, for instance:
+	 *  - add or remove attribute or class to an element,
+	 *  - change inside of the {@link module:engine/view/uielement~UIElement UIElement},
+	 *  - mark some model elements to convert using {@link  module:engine/model/differ~Differ#refreshItem differ API}.
+	 *
+	 * Try to avoid changes which touch view structure:
+	 *  - you should not add or remove nor wrap or unwrap any view elements,
+	 *  - you should not change the editor data model in view post-fixer.
+	 *
+	 * If you need DOM elements to be already updated, use {@link module:engine/view/view~View#event:render render event}.
 	 *
 	 * As a parameter, a post-fixer callback receives a {@link module:engine/view/downcastwriter~DowncastWriter downcast writer}
 	 * instance connected with the executed changes block.
 	 *
-	 * Note that registering a post-fixer won't re-render the editor's view. If the view should change after registering the post-fixer then
-	 * it should be done manually calling `view.forceRender();`.
+	 * Here is an example from the link plugin which adds highlight class to the link if the selection is in that link:
+	 *
+	 *		editor.editing.view.document.registerPostFixer( writer => {
```

Comment:
This code snippet is incomplete. What is `findLinkRange()`? And what does `highlightedLinks` do?

Reply:
It's fine to have some pseudo-code-like example, but it must be clear what it does. It's actually not clear to me how this one works.

Reply:
I removed `highlightedLinks`, it should not be there.

However, I believe we should not put the code of `findLinkRange` here. It is pretty complicated and would make the whole snippet less readable.

It is not easy to find a good example of the `registerPostFixer` function. Usually, it is part of a bigger picture.

Reply:
I do not know how I should describe what does `findLinkRange` do. For me, the name tells pretty well what this function does.

Reply:
I'll try to figure something out regarding `findLinkRange()`. Probably link to its doc.